src/microbe_stage/CompoundBag.cs

```diff
@@ -78,7 +78,7 @@ public float AddCompound(Compound compound, float amount)
 
         Compounds[compound] = newAmount;
 
-        return newAmount - amount;
+        return newAmount - existingAmount;
```

Comment:
I very strongly think that this will have a bunch of effects elsewhere. I think the return value was changed for colonies to work correctly. So changing the return value here is very likely to cause issues.

Reply:
As I wrote, the only possible effects are for colonies: it is called 9 times, 6 of which are calls without storing the variable. Out of the 3 remaining calls, 2 are from `MicrobeInterior.cs`, with no effect, as you can see right bellow

```
                var amountAdded = copyEntity.Compounds.AddCompound(compound, amount * 0.5f);

                if (amountAdded < some_var)
                {
                    // TODO: handle the excess compound that didn't fit in the other cell
                }
         } // exit function, the var was local
```

The last call is used in `ColonyCompoundBag.cs`, as follows:
```
public float AddCompound(Compound compound, float amount)
    {
        var originalAmount = amount;

        foreach (var bagToAddTo in GetCompoundBags())
        {
            var amountAdded = bagToAddTo.AddCompound(compound, amount);
            amount -= amountAdded;

            if (amount <= MathUtils.EPSILON)
                break;
        }

        return originalAmount - amount;
    }
```
As I said, the logic seems to use the amount added (which is what is documented and what I changed the return to), and not the previous amount (which was the previous logic). But you are right the the return is wrong again...

Reply:
For testing simplicity, I would *really* prefer if you could do the required math in the chunk code and not change the compound bag code as that is a truly core component. It took a lot of convincing of me for the colony code to be allowed to change the return value of that method.

Reply:
Actually, the colony compound bag return doesn't seem to be used elsewhere either. I still fixed the logic there.

Reply:
The issue is that the return value of the compound bag code is said to be precisely what I changed it to! And all occurences of named variable take into account that this *should* return the added value, yet it isn't.

To me this is clearly a bug in the code, that may mess up things later on (if it didn't already).

Reply:
Okay, I looked through the code base and this seems like a safe change.